builtin/providers/azurerm/resource_arm_network_interface.go

```diff
@@ -0,0 +1,393 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmNetworkInterface returns the *schema.Resource
+// associated to network interface resources on ARM.
+func resourceArmNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNetworkInterfaceCreate,
+		Read:   resourceArmNetworkInterfaceRead,
+		Update: resourceArmNetworkInterfaceCreate,
+		Delete: resourceArmNetworkInterfaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vm_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"mac_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"network_security_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"ip_config": &schema.Schema{
+				Type:     schema.TypeList,
```

Comment:
Also, please note that the existence of this `id` field for these mandatorily nested sub-resources makes using a *schema.Set here lacking a lot in usability.
As such; I'd argue that the ordered indexing of *schema.TypeList is needed here...